pkg/providers/amifamily/ami.go

```diff
@@ -79,25 +79,25 @@ func (a AMIs) Sort() {
 	})
 }
 
-func (a AMIs) String() string {
+func PrettyFormat(amis []v1beta1.AMI) string {
 	var sb strings.Builder
-	ids := lo.Map(a, func(a AMI, _ int) string { return a.AmiID })
-	if len(a) > 25 {
+	ids := lo.Map(amis, func(a v1beta1.AMI, _ int) string { return a.ID })
+	if len(amis) > 25 {
 		sb.WriteString(strings.Join(ids[:25], ", "))
-		sb.WriteString(fmt.Sprintf(" and %d other(s)", len(a)-25))
+		sb.WriteString(fmt.Sprintf(" and %d other(s)", len(amis)-25))
 	} else {
 		sb.WriteString(strings.Join(ids, ", "))
 	}
 	return sb.String()
 }
 
 // MapToInstanceTypes returns a map of AMIIDs that are the most recent on creationDate to compatible instancetypes
-func (a AMIs) MapToInstanceTypes(instanceTypes []*cloudprovider.InstanceType) map[string][]*cloudprovider.InstanceType {
+func MapToInstanceTypes(instanceTypes []*cloudprovider.InstanceType, amis []v1beta1.AMI) map[string][]*cloudprovider.InstanceType {
```

Comment:
nit: I'd expect that the first argument to a function that's called "MapToInstanceTypes" is the thing that we're mapping. I'd re-order these arguments so that amis come first